crates/starknet-devnet/tests/test_abort_blocks.rs

```diff
@@ -233,9 +291,37 @@ mod abort_blocks_tests {
 
         let fork_block_hash = fork_devnet.create_block().await.unwrap();
 
-        let aborted_blocks = abort_blocks(&fork_devnet, &fork_block_hash).await;
+        let aborted_blocks = abort_blocks(&fork_devnet, &BlockId::Hash(fork_block_hash)).await;
         assert_eq!(aborted_blocks, vec![fork_block_hash]);
 
-        abort_blocks_error(&fork_devnet, &fork_block_hash).await;
+        abort_blocks_error(&fork_devnet, &BlockId::Hash(fork_block_hash)).await;
+    }
+
+    #[tokio::test]
+    async fn abort_latest_blocks() {
+        let devnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", "full"])
+                .await
+                .expect("Could not start Devnet");
+
+        for _ in 0..3 {
+            devnet.create_block().await.unwrap();
+        }
+        for _ in 0..3 {
+            abort_blocks(&devnet, &BlockId::Tag(BlockTag::Latest)).await;
+        }
+        abort_blocks_error(&devnet, &BlockId::Tag(BlockTag::Latest)).await; // Rolled back to genesis block, should not be possible to abort
+    }
+    #[tokio::test]
+    async fn abort_pending_block() {
+        let devnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", "full"])
+                .await
+                .expect("Could not start Devnet");
+
+        for _ in 0..3 {
+            devnet.create_block().await.unwrap();
+        }
+        abort_blocks_error(&devnet, &BlockId::Tag(BlockTag::Pending)).await;
```

Comment:
Don't we want the pending block to be abortable?

I think proper testing of pending block abortion should be done in `--block-generation-on demand` mode. We want to assert that aborting the pending block removes any pending transactions, i.e. that the state is the same as on the latest block.

E.g. we could mint in on-demand mode, check that the pending balance is increased, then abort the block and check that the pending balance is reverted.

cc: @mikiw

Reply:
Aha, so idea is to just mine pending block internally, and then abort it normally? I was thinking maybe it's better to force user to do that, but I can't see now why.

Reply:
> Don't we want the pending block to be abortable?

Yes!

> I think proper testing of pending block abortion should be done in `--block-generation-on demand` mode.

Yes!

> E.g. we could mint in on-demand mode, check that the pending balance is increased, then abort the block and check that the pending balance is reverted.

Yes! Good Idea!